recipes/armadillo/all/conanfile.py

```diff
@@ -271,10 +306,24 @@ def package(self):
         copy(self, "NOTICE.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
         rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         rmdir(self, os.path.join(self.package_folder, "share"))
+        self._create_cmake_module_variables(os.path.join(self.package_folder, self._module_vars_rel_path))
+
 
     def package_info(self):
         self.cpp_info.libs = ["armadillo"]
         self.cpp_info.set_property("pkg_config_name", "armadillo")
+        self.cpp_info.set_property("cmake_find_mode", "both")
+        self.cpp_info.set_property("cmake_file_name", "Armadillo")
+        self.cpp_info.set_property("cmake_target_name", "Armadillo::Armadillo")
```

Comment:
I'll be happy to approve your PR but I wanted to understand first what is the logic for setting the `cmake_file_name` and `cmake_target_name` ?

I generally see it lowercase in other recipe ? Is it a convention from the conan-center-index or something related to the upstream repository ?

Reply:
Good question. The file name is being overridden here to match the upstream cmake module, which is named FindArmadillo.cmake. The target, since it's inherited from the module name, has also been overridden so that it's not interpreted as `armadillo::armadillo`. As per my comment in https://github.com/conan-io/conan-center-index/issues/20067#issuecomment-1839749295, I'm a little uncertain as to what to make the target name as it's not defined upstream. However, a couple of data points:

* the [`ensmallen` library uses `Armadillo::Armadillo`](https://github.com/mlpack/ensmallen/blob/5bb9fac6cf543ded1c69993a49b6fad285eef356/CMakeLists.txt#L61) (maintained by the same author as armadillo)
* The [`carma` library uses `armadillo::armadillo`](https://github.com/RUrlus/carma/blob/1a984a44e849fa632315d196dd71f8f29e670534/CMakeLists.txt#L73) (also maintained by the same author as armadillo), but they define their own [`GetArmadillo.cmake`](https://github.com/RUrlus/carma/blob/stable/cmake/GetArmadillo.cmake)
* [vcpkg uses `armadillo` as the target](https://github.com/Sentinel-One/vcpkg/blob/2b1757dd1d5ed7d9d5bb65fd8316631b157e7afc/ports/armadillo/usage#L4)
* UncommonCMakeModules (unofficial) [uses `Armadillo::Armadillo`](https://github.com/markjolah/UncommonCMakeModules/blob/f55b3ec18868c8ccfdd68ec4333c4e3b963e618d/FindArmadillo.cmake#L43)

[GitHub search results](https://github.com/search?q=%2Ftarget_link_libraries%5C%28.*%28PUBLIC%7CPRIVATE%7CINTERFACE%29+Armadillo%2F&type=code)

So, there's a bit of a smattering and no clear example of which target to choose. My thoughts are that `Armadillo::Armadillo` is the most consistent target, especially given usage in `ensmallen` and the module name. Having said that, I'm keen to hear community/team suggestions on consensus.

Reply:
So how about an additional `self.cpp_info.set_property("cmake_target_aliases", ["armadillo::armadillo"])` as a compromise?

Reply:
Great idea, thanks @valgur

Reply:
I've updated the recipe with four different aliases to provide reasonable coverage of possible existing usage.